src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1130,7 +1119,12 @@ public static bool IsLeapYear(int year)
             {
                 ThrowHelper.ThrowArgumentOutOfRange_Year();
             }
-            return (year & 3) == 0 && ((year & 15) == 0 || (year % 25) != 0);
+            if ((year & 3) != 0) return false;
+            if ((year & 15) == 0) return true;
+            // return true/false not the test result https://github.com/dotnet/runtime/issues/4207
+            if (!Bmi2.IsSupported) return (uint)year % 25 != 0 ? true : false;
+            // 42949673 = compute_unsigned_magic_info(100, 14) from https://github.com/ridiculousfish/libdivide/blob/master/doc/divide_by_constants_codegen_reference.c
+            return (uint)year != Bmi2.MultiplyNoFlags((uint)year, 42949673) * 100 ? true : false;
```

Comment:
This is just multiplying two uints. Why does this need to use hardware intrinsics?

Reply:
The default multiplication by reciprocal needs to account for a 32-bit uint, but by knowing that he input is limited to 14 bits (9999), it's possible to achieve it with a single multiplication (minus all the shifts/subs). Using MULX here to get the high 32-bits of the multiplication.

Reply:
Ah, ok. Is this tricky code really adding significant benefits over what the JIT does for you already?

Reply:
isn't `(uint)(year * -1030792151) > 171798691)` more efficient than this BMI2 code?